pkg/extension/workspace/toolchain/builtin/toolchain.go

```diff
@@ -39,7 +39,7 @@ var (
 )
 
 // NewToolchain returns a builtin toolchain with a given configuration.
-func NewToolchain(name string, cfg *config.ToolchainConfig, workspace model.Workspace) *builtin {
+func NewToolchain(name string, cfg *config.ToolchainConfig, workspace model.Workspace) *builtin { //nolint
```

Comment:
the lint error is that returning unexported types is confusing. I feel it probably isn't unexpected here, so disabled it.